engine.js

```diff
@@ -31,6 +31,25 @@ module.exports = function(options) {
   var getFromOptionsOrDefaults = function(key) {
     return options[key] || defaults[key];
   };
+  var getJiraIssueLocation = function(location, type, scope, jiraWithDecorators, subject) {
+    var commitMessage = ''
+
+    switch(location) {
+      case 'pre-type':
+        commitMessage = jiraWithDecorators + type + scope + ': ' + subject;
+        break;
+      case 'pre-description':
+        commitMessage = type + scope + ': ' + jiraWithDecorators + subject;
+        break;
+      case 'post-description':
+        commitMessage = type + scope + ': ' + subject + ' ' + jiraWithDecorators;
+        break;
+      default:
+        commitMessage = type + scope + ': ' + jiraWithDecorators + subject;
+    }
+
+    return commitMessage;
```

Comment:
What do you think about an early return? That way there is less mutability and the statements get shorter (as break is not necessary).

```suggestion
    switch(location) {
      case 'pre-type':
        return commitMessage = jiraWithDecorators + type + scope + ': ' + subject;
      case 'pre-description':
        return commitMessage = type + scope + ': ' + jiraWithDecorators + subject;
      case 'post-description':
        return commitMessage = type + scope + ': ' + subject + ' ' + jiraWithDecorators;
      default:
        return commitMessage = type + scope + ': ' + jiraWithDecorators + subject;
    }
```

Reply:
Sounds good! But we can also remove the commitMessage var and just return the result, right?